realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -75,6 +84,82 @@ static RealmCacheType valueOf(Class<? extends BaseRealm> clazz) {
         }
     }
 
+    private static class CreateRealmRunnable<T extends BaseRealm> implements Runnable {
+        private RealmConfiguration configuration;
+        private RealmInstanceCallback<T> callback;
+        private Class<T> realmClass;
+        private CountDownLatch fgRealmCreatedLatch = new CountDownLatch(1);
+        private RealmNotifier notifier;
+
+        CreateRealmRunnable(RealmNotifier notifier, RealmConfiguration configuration,
+                            RealmInstanceCallback<T> callback, Class<T> realmClass) {
+            this.configuration = configuration;
+            this.realmClass = realmClass;
+            this.callback = callback;
+            this.notifier = notifier;
+        }
+
+        @Override
+        public void run() {
+            T instance = null;
+            final AtomicBoolean taskCanceled = new AtomicBoolean(false);
+            try {
+                instance = createRealmOrGetFromCache(configuration, realmClass);
+                boolean results = notifier.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        // If the RealmAsyncTask.cancel() is called before, we just return without creating the Realm
+                        // instance on the caller thread.
+                        // Thread.isInterrupted() cannot be used for checking here since CountDownLatch.await() will
+                        // will clear interrupted status.
+                        if (taskCanceled.get()) {
+                            return;
+                        }
+                        T instanceToReturn = null;
+                        try {
+                            instanceToReturn = createRealmOrGetFromCache(configuration, realmClass);
+                        }catch (RuntimeException e) {
```

Comment:
Catch `Throwable` instead, otherwise some errors will not propagate to the Looper thread?